### ISSUE TITLE
Login promise error

### ISSUE THREAD

**Issue by dualshote** [jeduan/cordova-plugin-facebook4#649]
Hi,
I'm using the Cordova plugin facebook since five months ago, and i never had problem with it. 

But now, the login method return everytime an empty error object... I don't understand and i don't find any solution on internet :/ 

Some one can help me ? :) 
Thanks !

**Comment by peterpeterparker** [comment]
If I understand correctly you don't face an error but get an empty result from Facebook. In that case the problem isn't related to the plugin.

Plz use the Ionic forum or Stackoverflow channels for support, thx in advance

Side note: If you query for example education and work as permissions, it's normal that you get now empty results. These permissions were shortly deprecated from Facebook and are now returned as empty. See Facebook developers blog for more details

**Comment by dualshote** [comment]
Hi @peterpeterparker , i don't understand your answer... I don't find any solution on internet. Work perfectly on IOS, but on Android i have an empty error with the Login function :( 
What do you mean with : "If you query for example education and work as permissions, it's normal that you get now empty results"

Can you look at my code ? I really don't understand.... Thx ! 

<img width="922" alt="capture d ecran 2018-04-25 a 16 13 33" src="https://user-images.githubusercontent.com/5940350/39251508-c2d320c8-48a3-11e8-8bce-b8614b5f6ea0.png">

**Comment by peterpeterparker** [comment]
@dualshote you could forget my side note about education and work, you don't seems to use about these permissions

about support, because it still looks like a problem related to your particular app, my comment is still valid, you will have more chances to find support on stackoverflow or ionic forum

that being said, firstly, your code seems a bit weird no? you are using a typescript function `=>` for the success login but `function (...` for error handling and api request. Shouldn't all the code be written with `=>` ?

     this.fb.login(permissions).then((response: any) => {
     }, (error) => {
     });

secondly do you perform a `getLoginStatus` before doing your `login`? have you try with an icognito browser or logout yourself from Facebook first before doing your test again?

     this.fb.getLoginStatus().then((success: any) => {
          if (success.status === 'connected') {
              // No login
          } else {
             // Do login
          }
     }, (err) => {
         // error 
      });
